services/controller/standard/events.go

```diff
@@ -332,3 +337,41 @@ func (s *Service) subscribeToBeaconCommittees(ctx context.Context,
 	s.subscriptionInfos[epoch] = subscriptionInfo
 	s.subscriptionInfosMutex.Unlock()
 }
+
+// VerifySyncCommitteeMessages handles the "head" events from the beacon node for sync committee validation.
+func (s *Service) VerifySyncCommitteeMessages(ctx context.Context, data *apiv1.HeadEvent) {
+	_, span := otel.Tracer("attestantio.vouch.services.controller.standard").Start(ctx, "VerifySyncCommitteeMessages")
+	defer span.End()
+
+	logger := log.With().Uint64("slot", uint64(data.Slot)).Logger()
+	logger.Trace().Msg("Received head event")
+
+	rootReported, found := s.syncCommitteeMessenger.GetBeaconBlockRootReported(data.Slot)
+	if !found {
+		logger.Trace().Msg(fmt.Sprintf("no reported sync committee message data for slot: %d, skipping validation", data.Slot))
+		return
+	}
+
+	if rootReported.Root != data.Block {
+		logger.Trace().Msg(fmt.Sprintf("mismatch in block root for slot: %d. Reported: %s and observed: %s", data.Slot, rootReported.Root.String(), data.Block.String()))
+		// TODO: Check that it is correct to return here. If we do not match the block root we shouldn't be part of the SyncAggregate
+		return
+	}
+	blockResponse, err := s.signedBeaconBlockProvider.SignedBeaconBlock(ctx, &api.SignedBeaconBlockOpts{
+		Block: data.Block.String(),
+	})
+	if err != nil {
+		logger.Trace().Msg(fmt.Sprintf("failed to retrieve block: %s for slot: %d with err: %v", data.Block.String(), data.Slot, err))
+		return
+	}
+	syncAggregate, err := blockResponse.Data.SyncAggregate()
+	if err != nil {
+		logger.Trace().Msg(fmt.Sprintf("failed to get sync aggregate for block: %s for slot: %d with err: %v", data.Block.String(), data.Slot, err))
+		return
+	}
+	for _, index := range rootReported.ValidatorIndices {
```

Comment:
There is a mismatch here.  The sync committee bits are for the sync committee, not the validators themselves.  In other words `SyncCommitteeBits` is always 512 bits long, regardless of how many validators we have.

The sync committee duty provides the mapping from validator index to sync committee index.  This information is obtained elsewhere in the codebase, so may be able to be sneaked into existing structures rather than refetching it every time we are in this call.